integration/registry_test.go

```diff
@@ -0,0 +1,238 @@
+	_ "github.com/aquasecurity/fanal/analyzer/pkg/rpmcmd"
+)
+
+const (
+	registryImage    = "registry:2"
+	registryPort     = "5443/tcp"
+	registryUsername = "testuser"
+	registryPassword = "testpassword"
+)
+
+func TestTLSRegistry(t *testing.T) {
+	ctx := context.Background()
+
+	baseDir, err := filepath.Abs(".")
+	require.NoError(t, err)
+
+	req := testcontainers.ContainerRequest{
+		Name:         "registry",
+		Image:        registryImage,
+		ExposedPorts: []string{registryPort},
+		Env: map[string]string{
+			"REGISTRY_HTTP_ADDR":            "0.0.0.0:5443",
+			"REGISTRY_HTTP_TLS_CERTIFICATE": "/certs/cert.pem",
+			"REGISTRY_HTTP_TLS_KEY":         "/certs/key.pem",
+			"REGISTRY_AUTH":                 "htpasswd",
+			"REGISTRY_AUTH_HTPASSWD_PATH":   "/auth/htpasswd",
+			"REGISTRY_AUTH_HTPASSWD_REALM":  "Registry Realm",
+		},
+		BindMounts: map[string]string{
+			filepath.Join(baseDir, "data", "registry", "certs"): "/certs",
+			filepath.Join(baseDir, "data", "registry", "auth"):  "/auth",
+		},
+		WaitingFor: wait.ForLog("listening on [::]:5443"),
+	}
+
+	registryC, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+	require.NoError(t, err)
+	defer registryC.Terminate(ctx)
+
+	registryURL, err := getRegistryURL(ctx, registryC, registryPort)
+	require.NoError(t, err)
+
+	config := testdocker.RegistryConfig{
+		URL:      registryURL,
+		Username: registryUsername,
+		Password: registryPassword,
+	}
+
+	testCases := []struct {
+		name       string
+		imageName  string
+		imageFile  string
+		option     types.DockerOption
+		login      bool
+		expectedOS types.OS
+		wantErr    bool
+	}{
+		{
+			name:      "happy path",
+			imageName: "alpine:3.10",
+			imageFile: "testdata/fixtures/alpine-310.tar.gz",
+			option: types.DockerOption{
+				Timeout:               60 * time.Second,
+				UserName:              registryUsername,
+				Password:              registryPassword,
+				InsecureSkipTLSVerify: true,
+			},
+			expectedOS: types.OS{Name: "3.10.2", Family: "alpine"},
+			wantErr:    false,
+		},
+		{
+			name:      "happy path with docker login",
+			imageName: "alpine:3.10",
+			imageFile: "testdata/fixtures/alpine-310.tar.gz",
+			option: types.DockerOption{
+				Timeout:               60 * time.Second,
+				InsecureSkipTLSVerify: true,
+			},
+			login:      true,
+			expectedOS: types.OS{Name: "3.10.2", Family: "alpine"},
+			// TODO: this should be false, but there is regression now.
+			// After replacing containers/image with google/go-containerregistry, it is supposed to pass.
+			wantErr: true,
+		},
+		{
+			name:      "sad path: tls verify",
+			imageName: "alpine:3.10",
+			imageFile: "testdata/fixtures/alpine-310.tar.gz",
+			option: types.DockerOption{
+				Timeout:  60 * time.Second,
+				UserName: registryUsername,
+				Password: registryPassword,
+			},
+			expectedOS: types.OS{Name: "3.10.2", Family: "alpine"},
```

Comment:
No need to specify OS expectation as you expect err?

Reply:
Thanks. I removed it.
https://github.com/aquasecurity/fanal/pull/99/commits/dc13b099a7934c42daba188eebe84b1bf8c01851